envoy/upstream/cluster_manager.h

```diff
@@ -214,6 +214,18 @@ class ClusterManager {
 
   virtual ~ClusterManager() = default;
 
+  // API to initializes the ClusterManagerImpl instance based on the given Bootstrap config.
+  //
+  // This method *must* be called prior to invoking any other methods on the class and *must* only
+  // be called once. This method should be called immediately after ClusterManagerImpl construction
+  // and from the same thread in which the ClusterManagerImpl was constructed.
+  //
+  // The initialization is separated from the constructor because lots of work, including ADS
+  // initialization, is done in this method. If the contents of this method are invoked during
+  // construction, a derived class cannot override any of the virtual methods and have them invoked
+  // instead, since the base class's methods are used when in a base class constructor.
+  virtual absl::Status init(const envoy::config::bootstrap::v3::Bootstrap& bootstrap) PURE;
```

Comment:
nit: init() -> initialize()

And I think maybe we could also provide an additional initialized() method to tell if the cluster_manager is initialized?

Reply:
done